packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -3666,6 +3666,14 @@ export class Checker extends ParseTreeWalker {
         if (this._fileInfo.diagnosticRuleSet.reportShadowedImports === 'none') {
             return;
         }
+
+        // Skip this check for relative imports.
+        const nodeModule = node.nodeType === ParseNodeType.ImportFromAs ? (node.parent as any)?.module : node.module;
```

Comment:
This isn't the right way to test for this. You should see if the parent of the ImportFromAsNode is an ImportFromNode. If so, then you can check whether the `module.leadingDots.length > 0` to see if it's a relative import. 

What you're doing here is directly accessing the analyzerNodeInfo without going through the proper accessor function (`getImportInfo`). There's no need to use that information here since the parser output contains the necessary information.